test/test-PerfMeter.js

```diff
@@ -0,0 +1,162 @@
+/* globals Services */
+
+"use strict";
+const simplePrefs = require("sdk/simple-prefs");
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const {setTimeout} = require("sdk/timers");
+const {ActivityStreams} = require("lib/ActivityStreams");
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/Services.jsm");
+
+let ACTIVITY_STREAMS_URL;
+let app;
+
+function openTestTab(notifyEvent) {
+  let tabData = {};
+  tabData.showPromise = new Promise(resolve => {
+    function onShow(tab) {
+      tabData.tab = tab;
+      tabs.removeListener("pageshow", onShow);
+      if (!notifyEvent) {
+        setTimeout(function() {
+          resolve();
+        });
+      }
+    }
+
+    let observer = {
+      observe: function(subject, topic, data) {
+        Services.obs.removeObserver(this, notifyEvent);
+        setTimeout(function() {
+          resolve(JSON.parse(data));
+        });
+      }
+    };
+
+    if (notifyEvent) {
+      Services.obs.addObserver(observer, notifyEvent);
+    }
+    tabs.on("pageshow", onShow);
+  });
+  tabs.open(ACTIVITY_STREAMS_URL);
+  return tabData;
+}
+
+exports.test_PerfMeter_init = function(assert) {
+  assert.deepEqual(app.performanceData, {}, "no performance data");
+};
+
+exports.test_PerfMeter_events = function*(assert) {
+  let exectedEvents = new Set([
+    "TAB_OPEN",
+    "WORKER_ATTACHED",
+    "TAB_READY",
+    "TOP_FRECENT_SITES_REQUEST",
+    "RECENT_LINKS_REQUEST",
+    "FRECENT_LINKS_REQUEST",
+    "RECENT_BOOKMARKS_REQUEST",
+    "SEARCH_STATE_REQUEST",
+    "NOTIFY_PERFORMANCE",
+    "TOP_FRECENT_SITES_RESPONSE",
+    "RECENT_LINKS_RESPONSE",
+    "RECENT_BOOKMARKS_RESPONSE",
+    "SEARCH_STATE_RESPONSE",
+    "NOTIFY_PERFORMANCE",
+  ]);
+
+  let tabData = openTestTab("performance-log-complete");
+  let notifyData = yield tabData.showPromise;
+
+  let performanceData = app.performanceData;
+  assert.ok(performanceData[tabData.tab.id], "Entry with expected tab id exists");
+
+  let events = performanceData[tabData.tab.id].events;
+  for (let i = 0; i < exectedEvents.length; i++) {
+    assert.ok(exectedEvents.has(events[i].tag), "Received expected event " + exectedEvents[i]);
```

Comment:
This ensures that the events recorded by perfMeter are what we expect them to be.